### PR TITLE
Show E Can Gains on Items Page

### PR DESCRIPTION
**Pull request by Sashank999**
Title says it all.
Any edits are welcome and I request to do soome additional testing who has some perks from company and faction which increase E Can gains.

### CODE REVIEW

File: torntools/scripts/content/items/ttItems.js [hunk 1]
```diff
@@ -757,3 +758,22 @@ function showMissingFlowers() {
 		}
 	}
 }
+
+function showECanGains() {
+	let facECanPerc = parseInt(userdata.faction_perks.filter((x) => /energy drinks/i.test(x)).map((x) => {
+		x.replace(/[^0-9\.]/g, "");
+	})[0]);
```
Comment (Mephiles) [inline torntools/scripts/content/items/ttItems.js:765]: Could you add a comment in code what `/energy drinks/i.test(x)` and `x.replace(/[^0-9\.]/g, "")` do. Just for future reference as I don't know what the first one does and I don't want to crack my head in reading regex 😄 .

File: torntools/scripts/content/items/ttItems.js [hunk 2]
```diff
@@ -757,3 +758,22 @@ function showMissingFlowers() {
...
+	let jobECanPerc = parseInt(userdata.company_perks.filter((x) => /boost/i.test(x)).map((x) => {
+		x.replace(/[^0-9\.]/g, "");
+	})[0]);
+	doc.findAll("[data-category='Energy Drink']").forEach((ECan) => {
```
Comment (Mephiles) [inline torntools/scripts/content/items/ttItems.js:769]: I'd say this goes against our variable naming convention. 1. this variable starts with a capital letter and 2. this variable represents an element, so it's more readable if the name depicts it.

For example `energyDrinkElement` or `energyDrinkRow`, etc.

File: torntools/scripts/content/items/ttItems.js [hunk 3]
```diff
@@ -757,3 +758,26 @@ function showMissingFlowers() {
...
+		// Replace everything other than numbers
+		x.replace(/[^0-9\.]/g, "");
+	})[0]);
+	doc.findAll("[data-category='Energy Drink']").forEach((eCanElement) => {
+		if (!eCanElement.find("span#tt-e-can")) {
+			let baseE = parseInt(itemlist.items[eCanElement.getAttribute("data-item")].effect.split(" ").map((x) => parseInt(x)).filter((x) => !isNaN(x))[0]);
+			let totalEnergy = baseE;
+			if (!isNaN(facECanPerc)) totalEnergy += (facECanPerc / 100) * baseE;
+			if (!isNaN(jobECanPerc)) totalEnergy += (jobECanPerc / 100) * baseE;
+			rawHTML = `<span id='tt-e-can' class='tt-item-price' style='padding-left: 5px;float: none;'>${totalEnergy}E</span>`;
```
Comment (DeKleineKobini) [inline torntools/scripts/content/items/ttItems.js:779]: You shouldn't use an id here, as it won't be unique. Instead, add it as class and add the inline css to the `ttItems.css` for that class.

File: torntools/scripts/content/items/ttItems.js [hunk 4]
```diff
@@ -757,3 +758,26 @@ function showMissingFlowers() {
...
+			eCanElement.find("span.qty.bold.t-hide").insertAdjacentHTML("afterEnd", rawHTML);
```
Comment (DeKleineKobini) [inline torntools/scripts/content/items/ttItems.js:780]: `.bold.t-hide` shouldn't be checked here. These are used for css by Torn and we should only use them as last resort.

**Review by DeKleineKobini — APPROVED** [review]

**Pull request merged**